### ISSUE TITLE
Deploying to heroku problems

### ISSUE THREAD

**Issue by rpbaltazar** [whatcould/webshims-rails#19]
I've followed the Readme and opted by following option 1.
I've added the task to my deploy script (which is running) and updated the path to /webshims/shims/.

Everything seems to work locally, as well as in my CI machine.

2 issues appear on Heroku, though:

1- The assets are not being served from public/webshims/shims.
    I solved this by changing back the path to assets/webshims/shims.
2- The following error appears in the console:
```
Modernizr was included more than once. Make sure to include it only once! Webshims and other scripts might not work properly.
```
   Removing *require modernizr* from my application.js doesn't solve the issue either because then Modernizr is not defined.

**Comment by whatcould** [comment]
Hmm, seems like there are two separate issues; public/webshims/shims not working, and the modernizr duplication.

When you say "added the task to my deploy script", what does that mean? Are you running the rake task on heroku (`heroku run rake xxx`)? And "assets are not being served", you mean, a request to yourapp/webshims/shims/1.js is a 404? I'm not sure how the heroku filesystem works as far as the rake task — I don't know if it allows you to dynamically add files to the slug like that. I've just committed these files to my application's repository.

The modernizr issue, let me think about a minute. It does seem like maybe you're including it twice in your application.js, though that seems like an obvious answer.

**Comment by rpbaltazar** [comment]
I have a script that does a bunch of stuff on the heroku (building cache, ...).
The rake task does not run properly by doing ```heroku run rake webshims:update_public```. It does not create the folder in the slug. Only if I connect to the heroku bash and run it manually, then it creates the folder with the shims, but still the assets are not served properly.
Today I'll do some more tests to try to debug the problem more clearly.

For the modernizr I'm taking a second look at it. Seems that somehow heroku is picking up a old version of combo/10.js and that is causing the problem.

**Comment by rpbaltazar** [comment]
I solved the issue for now.
So, as I said before, running the task through ```heroku run xxx``` does not create the folder in heroku, for some reason. (I'm assuming heroku does not give you permission for that).

The solution I found, passes by
1. run the task in your local machine
2. add the public/webshims folder to your repo
3. publish to heroku
Note: using rails_12factor which allows to serve static assets.

The problem related with Modernizr was because for some reason, heroku was loading the old webshims we had in the repo previously, which were trying to load and reload the Modernizr, so nothing related with this gem nor the webshims themselves.
